### PR TITLE
Fix eslint warnings

### PR DESCRIPTION
**Pull request by ofpiyush**
## Description
Eslint and subsequently codefactor keep warning/erroring on issues. This PR fixes all current eslint warnings from local

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manually. `yarn start` does not generate warnings.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/propertyControls/index.test.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 // TODO Abhinav, this is not passing
 // import { getPropertyControlTypes } from "./index";
+// TODO: Is this required?
+// eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/components/propertyControls/index.test.ts:4]: You can remove

File: app/client/src/components/stories/Menu.stories.tsx [hunk 2]
```diff
@@ -21,7 +21,8 @@ export default {
   component: Menu,
   decorators: [withKnobs, withDesign],
 };
-
+// TODO: Is this required?
+// eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/components/stories/Menu.stories.tsx:25]: @riodeuno @devrk96 Do we need this?

File: app/client/src/pages/Editor/Explorer/ExplorerTitle.tsx [hunk 3]
```diff
@@ -18,22 +18,11 @@ const ActionIconGroup = styled.div`
...
       <h1>EXPLORER</h1>
-      <ActionIconGroup>
-        {/* <Icon
-          iconSize={ICON_SIZE}
-          icon={props.isCollapsed ? "remove" : "add"}
-          onClick={props.onCollapseToggle}
-        />
-        <Icon icon="pin" iconSize={ICON_SIZE} /> */}
-      </ActionIconGroup>
+      <ActionIconGroup></ActionIconGroup>
```
Comment (hetunandu) [inline app/client/src/pages/Editor/Explorer/ExplorerTitle.tsx:25]: Close this tag properly

File: app/client/src/components/stories/Menu.stories.tsx (hunk 2 above)
Comment (devrk96) [reply app/client/src/components/stories/Menu.stories.tsx:25]: It is not required.

File: app/client/src/pages/common/AppRoute.tsx [hunk 4]
```diff
@@ -28,6 +28,7 @@ class AppRouteWithoutProps extends React.Component<AppRouteProps> {
   }
 
   render() {
+    // eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/pages/common/AppRoute.tsx:31]: Remove component from deconstruction and spread the rest

File: app/client/src/pages/common/CustomizedDropdown/OrgDropdownData.tsx [hunk 5]
```diff
@@ -8,6 +8,8 @@ import { Org } from "constants/orgConstants";
 import { User } from "constants/userConstants";
 import _ from "lodash";
 
+// TODO: is this required?
+// eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/pages/common/CustomizedDropdown/OrgDropdownData.tsx:12]: @riodeuno Is this still required

File: app/client/src/reducers/uiReducers/applicationsReducer.tsx [hunk 6]
```diff
@@ -26,6 +26,7 @@ const initialState: ApplicationsReduxState = {
 const applicationsReducer = createReducer(initialState, {
   [ReduxActionTypes.DELETE_APPLICATION_INIT]: (
     state: ApplicationsReduxState,
+    // eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/reducers/uiReducers/applicationsReducer.tsx:29]: Remove action arg from here and define the signature in the action dispatcher function

File: app/client/src/reducers/uiReducers/applicationsReducer.tsx [hunk 7]
```diff
@@ -243,22 +245,21 @@ const applicationsReducer = createReducer(initialState, {
...
-            ...org.applications[appIndex],
-            ...rest,
-          };
-        }
+    const _organizations = state.userOrgs.map((org: Organization) => {
+      const appIndex = org.applications.findIndex(
+        (app) => app.id === action.payload.id,
+      );
+      // eslint-disable-next-line
+      const { id, ...rest } = action.payload;
```
Comment (hetunandu) [inline app/client/src/reducers/uiReducers/applicationsReducer.tsx:253]:
```suggestion
      const { id, ...rest } = action.payload;
      const appIndex = org.applications.findIndex(
        (app) => app.id === id,
      );
```

File: app/client/src/reducers/uiReducers/explorerReducer.ts [hunk 8]
```diff
@@ -98,13 +98,16 @@ const explorerReducer = createReducer(initialState, {
...
+    // eslint-disable-next-line
     state: ExplorerReduxState,
     action: ReduxAction<{ id: string }>,
   ) => {
     return { editingEntityName: action.payload.id };
   },
   [ReduxActionTypes.END_EXPLORER_ENTITY_NAME_EDIT]: (
+    // eslint-disable-next-line
     state: ExplorerReduxState,
+    // eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/reducers/uiReducers/explorerReducer.ts:110]: Can remove this

File: app/client/src/sagas/WidgetOperationSagas.tsx [hunk 9]
```diff
@@ -101,6 +101,7 @@ function getChildWidgetProps(
   const { leftColumn, topRow, newWidgetId, props, type } = params;
   let { rows, columns, parentColumnSpace, parentRowSpace, widgetName } = params;
   let minHeight = undefined;
+  // eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/sagas/WidgetOperationSagas.tsx:104]: disable just the rule

File: app/client/src/transformers/RestActionTransformer.ts [hunk 10]
```diff
@@ -41,6 +38,8 @@ export const transformRestAction = (data: any): any => {
         },
       );
       if (contentTypeHeader) {
+        // TODO: what is this logic doing here?
+        // eslint-disable-next-line
```
Comment (hetunandu) [inline app/client/src/transformers/RestActionTransformer.ts:42]: Mostly can be removed

**Review by hetunandu — APPROVED** [review]

**Pull request merged**